docs/the-new-architecture/turbo-modules-with-swift.md

```diff
@@ -0,0 +1,167 @@
+# iOS - Using Swift in Your Native Modules
+
+Swift is the official and default language for developing native application on iOS.
+
+In this guide, we will explore how you can write your Native Modules using Swift.
+
+:::note
+The core of React Native is mainly written in C++ and the interoperability between Swift and C++ is not great, despite the [interoperability layer](https://www.swift.org/documentation/cxx-interop/) developed by Apple.
+
+Therefore, the module you are going to write in this guide won't be a pure Swift implementation due to the incompatibilities between the languages. You'll have to write some Objective-C++ glue code but the goal of the guide is to minimize the amount of Objective-C++ code that is needed. If you are migrating an existing Native Modules from the legacy architecture to the New Architecture, this approach should allow you to reuse most of the code.
+:::
+
+This guide starts from the iOS implementation of the [Native Module](/docs/next/turbo-native-modules-introduction) guide.
+Make sure to be familiar with that guide before diving into this one, potentially implementing the example in the guide.
+
+## The Adapter pattern
+
+The goal we have is to implement all our business logic using a Swift module and have a thin glue layer in Objective-C++ that is able to connect the app with the Swift implementation.
+
+We can achieve this by leveraging the [Adapter](https://refactoring.guru/design-patterns/adapter) design pattern, to connect the Swift Module with the Objective-C++ layer.
```

Comment:
Maybe a wikipedia link is better? https://en.wikipedia.org/wiki/Adapter_pattern

Reply:
+1 to this